gass/copy/source/globus_gass_copy_glob.c

```diff
@@ -2148,16 +2157,34 @@ globus_l_gass_copy_cksm_file(
 
     mdctx = EVP_MD_CTX_create();
     EVP_DigestInit_ex(mdctx, md, NULL);
-
-    while((n = read(fd, buf, count)) > 0)
+    /** A recursive copy may attempt to perform a checksum on directory urls.
+     * To preserve behaviour with previous version return a no-data checksum
+     *
+     */
+    if( is_regular )
     {
-        if(length >= 0)
+        while((n = (readlen = read(fd, buf, count))) > 0)
```

Comment:
this is a bit of an odd construction: `n` and `readlen` are both set to the return value of read, why do you need both?

Reply:
probably don't need both.  However n is an int where readlen is the correct typedef for the return value. I wrote this while trying to debug some errors that turned out to be in the file-system and was getting paranoid